src/modules/authentication/components/Register/Register.tsx

```diff
@@ -1,7 +1,200 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { AUTH_URL, axiosInstance } from '../../../../services/EndPoints';
+import { toast } from 'react-toastify';
+import { Link, useNavigate } from 'react-router-dom';
+import { EMAIL_VALIDATION, passwordValidtion } from '../../../../services/validation';
+import logo from '../../../../assets/images/logo.png';
+
+
+
+interface IFormInput {
+  userName: string;
+  email: string;
+  country: string;
+  phoneNumber: number;
+  password: string;
+  confirmPassword: string;
+  profileImage: FileList;
+}
+
 export default function Register() {
+
+  const[isPasswordVisible, setIsPasswordVisible] =useState(false);
+  let navigate = useNavigate();
+
+  let {register,
+    formState:{errors},
+    handleSubmit,
+    
+  } = useForm<IFormInput>({mode:"onChange"});
+
+  
+  const onSubmit = async (data:any) => {
+
+    const formData = new FormData();
+      formData.append("userName", data?.userName);
+      formData.append("email", data?.email);
+      formData.append("country", data?.country);
+      formData.append("profileImage", data?.profileImage[0]);
+      formData.append("phoneNumber", data?.phoneNumber);
+      formData.append("password", data?.password);
+      formData.append("confirmPassword", data?.confirmPassword);
+
+
+    try {
+      let response:any =await axiosInstance.post(AUTH_URL.REGISTER,formData);
+      toast.success('please verify your Registeration');
```

Comment:
We can get the message coming from backend, and provide our message as a fallback.